packages/material-ui-lab/src/TabPanel/TabPanel.d.ts

```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import { StandardProps } from '@material-ui/core';
+import { BoxProps } from '@material-ui/core/Box';
+
+export type TabPanelClassKey = 'root';
+
+export interface TabPanelProps extends StandardProps<BoxProps, TabPanelClassKey> {
+  /**
+   * The currently active value. Must be the same value that is passed to `Tabs`.
+   */
+  activeValue?: unknown;
```

Comment:
Just curious if this can be made generic instead, with a fallback to `unknown`? Similar to https://github.com/downshift-js/downshift/blob/6714f554a3128e8a6ede3da86a99190727eb3b5e/typings/index.d.ts#L9

Reply:
Generics give a false sense of security in JSX since TypeScript can't type-check child-parent relationships. 

Making it generic would be the same as type casting. But with type casting you're aware that you're doing something a type-checker can't verify.

It's basically like doing `function foo<T>(): T`.

Reply:
> It's basically like doing `function foo<T>(): T`.

Ah is the main thing here that `foo` doesn't accept an argument that would be tied to the type parameter?

> Generics give a false sense of security in JSX since TypeScript can't type-check child-parent relationships.

Ah I didn't consider child-parent relationships, was more thinking about type inference in isolation, e.g. `<TabPanel activeValue={something} />` where `activeValue` would then be automatically typed as `typeof something`, but you're right `<TabPanel<DifferentType> activeValue={something} />` would be the same as type casting.

Reply:
I can take a look at it again but I'm fairly certain this is the same issue as with `Menu` or `Select`.

Reply:
By no means is this a blocker, was just curious really! I think that the parent-child stuff is probably why it can't be a generic.

Reply:
The new approach requires `string`. No need for generic or unknown types anymore.